subsys/logging/log_backend_spinel.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log_backend.h>
+#include <logging/log_output.h>
+#include "log_backend_std.h"
+
+#ifndef CONFIG_LOG_BACKEND_SPINEL_BUFFER_SIZE
+#define CONFIG_LOG_BACKEND_SPINEL_BUFFER_SIZE 0
+#endif
+
+static u8_t char_buf[CONFIG_LOG_BACKEND_SPINEL_BUFFER_SIZE];
+static bool panic_mode;
+
+static int write(u8_t *data, size_t length, void *ctx);
+
+LOG_OUTPUT_DEFINE(log_output, write,
+		  char_buf, sizeof(char_buf));
+
+static void put(const struct log_backend *const backend,
+		struct log_msg *msg)
+{
+	// prevent adding CRLF, which may crash spinel decoding
+	u32_t flag = LOG_OUTPUT_FLAG_CRLF_NONE;
+
+	log_backend_std_put(&log_output, flag, msg);
+}
+
+static void sync_string(const struct log_backend *const backend,
+		     struct log_msg_ids src_level, u32_t timestamp,
+		     const char *fmt, va_list ap)
+{
+	// prevent adding CRLF, which may crash spinel decoding
+	u32_t flag = LOG_OUTPUT_FLAG_CRLF_NONE;
+
+	log_backend_std_sync_string(&log_output, flag, src_level,
+				    timestamp, fmt, ap);
+}
+
+static void sync_hexdump(const struct log_backend *const backend,
+			 struct log_msg_ids src_level, u32_t timestamp,
+			 const char *metadata, const u8_t *data, u32_t length)
+{
+	// prevent adding CRLF, which may crash spinel decoding
+	u32_t flag = LOG_OUTPUT_FLAG_CRLF_NONE;
+
+	log_backend_std_sync_hexdump(&log_output, flag, src_level,
+				     timestamp, metadata, data, length);
+}
+
+static void log_backend_spinel_init(void)
+{
+	memset(char_buf, '\0', sizeof(char_buf));
+}
+
+static void panic(struct log_backend const *const backend)
+{
+	log_backend_std_panic(&log_output);
+	panic_mode = true;
+}
+
+static void dropped(const struct log_backend *const backend, u32_t cnt)
+{
+	ARG_UNUSED(backend);
+
+	log_backend_std_dropped(&log_output, cnt);
+}
+
+__attribute__((weak)) void otPlatLog(int aLogLevel, int aLogRegion,
```

Comment:
I'm afraid of this;) But leave it for now.

Reply:
Why do we need a weak definition in the first place?

Reply:
I did this, because I couldn't find clear way to include logging.h file, where otPlatLog method declaration is located. Moreover I thought that I shouldn't provide OpenThread specific includes inside very general Zephyr logging directory.

Reply:
I don't see an issue with introducing a header if we aleady use an OT function in the file, if that's the case. I also don't see why shouldn't it work (for instance `logging.c` use this header).

Nonetheless, I still don't quite understand why a weak definition is provided here instead of a function declaration:
`void otPlatLog(int aLogLevel, int aLogRegion, const char *aFormat, ...);` 
(which would not be needed if the header was provided).

Reply:
Hmm I did, as you said and it works. I'm not sure why I was having a problems. Probably I had some weird state because I messed up something in include paths or maybe switching between branches. Still have to learn project structure and using git...